docs/lib/graphqlapi/fragments/ios/query-data.md

```diff
@@ -121,3 +120,41 @@ func listTodos() -> AnyCancellable {
 </amplify-block>
 
 </amplify-block-switcher>
+
+> **Note**: This approach will only return up to the first 1,000 items.  To change this limit or make requests for additional results beyond this limit, use *pagination* as discussed below.
+
+## List subsequent pages of items
+
+A list query only returns the first 1,000 items by default, so for large data sets, you'll need to paginate through the results.  After receiving a page of results, you can check if there are subsequent pages and obtain the next page. The page size is configurable as well, as in the example below.
+
+
+```swift
+func listTodos() {
+    let todo = Todo.keys
+    let predicate = todo.name == "my first todo" && todo.description == "todo description"
+    Amplify.API.query(request: .paginatedList(Todo.self, where: predicate, limit: 1000)) { event in
```

Comment:
To reduce the nesting in this example, you could show this example using the Combine API. The various `switch`/`if` statements could be flattened to [`filter(_:)` calls](https://developer.apple.com/documentation/combine/publishers/merge/filter(_:)).

Reply:
thanks for the sample, i'll have to visit this usage again to be more familiar with it. In the latest revision, i took the approach to build a view model class that helps collect the next pages into an array. I've added the combine example, that only used for the first call, holds a reference to the `List<Todo>`, and uses it get check and get the next page